### ISSUE TITLE
Unit failures in CondTools/Ecal in 10_6_X IB

### ISSUE THREAD

**Issue by qliphy** [cms-sw/cmssw#33989]
As the title says, there are some unit failures:

https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc700/CMSSW_10_6_X_2021-06-05-1100/unitTestLogs/CondTools/Ecal#/106637-106637

----- Begin Fatal Exception 05-Jun-2021 17:29:20 CEST-----------------------
An exception of category 'ConditionDatabase' occurred while
   [0] Calling endJob for module ExTestEcalTPGWeightIdMapAnalyzer/'Test1'
Exception Message:
Payload of type EcalTPGWeightIdMap with id 16eed3e159c33063aec3598f6779d4632f39d848 could not be loaded. An exception of category 'ConditionDatabase' occurred.
Exception Message:
De-serialization failed: the current boost version (1_67) is unable to read the payload. Data might have been serialized with an incompatible version. Payload serialization info:  {
"CMSSW_version": "CMSSW_11_1_0",
"architecture": "slc7_amd64_gcc820",
"technology": "boost/serialization",
"tech_version": "1_72"
 }
 from default_deserialize 
 from Session::fetchPayload 
----- End Fatal Exception -------------------------------------------------

----- Begin Fatal Exception 05-Jun-2021 17:29:27 CEST-----------------------
An exception of category 'ConditionDatabase' occurred while
   [0] Calling endJob for module ExTestEcalDAQAnalyzer/'conf_o2o'
Exception Message:
Payload of type EcalCondTowerObjectContainer<EcalDAQStatusCode> with id 050701816cc5945c97587bfc8ded5bdb9d1ddef6 could not be loaded. An exception of category 'ConditionDatabase' occurred.
Exception Message:
De-serialization failed: the current boost version (1_67) is unable to read the payload. Data might have been serialized with an incompatible version. Payload serialization info:  {
"CMSSW_version": "CMSSW_11_1_0",
"architecture": "slc7_amd64_gcc820",
"technology": "boost/serialization",
"tech_version": "1_72"
 }
 from default_deserialize 
 from Session::fetchPayload 
----- End Fatal Exception -------------------------------------------------

----- Begin Fatal Exception 05-Jun-2021 17:29:37 CEST-----------------------
An exception of category 'ConditionDatabase' occurred while
   [0] Calling endJob for module EcalIntercalibConstantsPopConBTransitionAnalyzer/'popConEcalIntercalibConstants'
Exception Message:
Payload of type RunInfo with id 2a5b959ce054d381e14b71836c234b9561180d29 could not be loaded. An exception of category 'ConditionDatabase' occurred.
Exception Message:
De-serialization failed: the current boost version (1_67) is unable to read the payload. Data might have been serialized with an incompatible version. Payload serialization info:  {
"CMSSW_version": "CMSSW_11_1_0",
"architecture": "slc7_amd64_gcc820",
"technology": "boost/serialization",
"tech_version": "1_72"
 }
 from default_deserialize 
 from Session::fetchPayload 
----- End Fatal Exception -------------------------------------------------

**Comment by perrotta** [comment]
@cms-sw/db-l2 was there any progress in the identification of the reason for this issue?
Apparenty it shows up in 10_6, 11_1 and 11_2, but not in more recent releases

**Comment by ggovi** [comment]
The issue is well identified.  In 11_3  we have been switching to a new boost version. The corresponding payloads are no longer readable by the earlier releases like the ones listed. These tests, designed to validate the production workflow, are executed on a local test tag created copying the second-last iov from production tag. In this way, we simulate the ability to execute the last required update. The problem is, the last updates have been executed during Cruzet with the new boost! For this reason, the above listed releases are not able to read the concerned payloads. On the other hand, after the switch to 11_3 this validation is in principle no longer needed...
Bottom line, there are two options:  1. switch off these tests ( and apply this policy regularly while upgrading the online release ) after the switch to new boost on the online release 2. Create manually separate tags ( containing only the corresponding boost formats)  to be used as source for every release to be tested, and maintain them ( the matrix will grow with time... )
Please provide feedback!

**Comment by davidlange6** [comment]
Could tests instead be run with a stable GT?

**Comment by ggovi** [comment]
@davidlange6 are you referring to the snapshot mechanism, to ensure to always have a frozen input? Maybe we could, but
a) not sure if stable GTs (=with a specified snapshot ) exist including these tags b). we need to re-write the tests.

**Comment by davidlange6** [comment]
For example, yes - CI systems work best if they are reproducible and don't randomly break...

**Comment by ggovi** [comment]
An other proposal is offered by: https://github.com/cms-sw/cmssw/pull/35084
It looks it works for 11_2, if we are happy we extend it to the other branches where we have failures.

**Comment by ggovi** [comment]
Finally found a different approach:
https://github.com/cms-sw/cmssw/pull/35442

**Comment by tvami** [comment]
+db 
 * got resolved in https://github.com/cms-sw/cmssw/pull/35442
